operations/app/src/modules/application_insights/alert_db.tf

```diff
@@ -0,0 +1,114 @@
+# Severity 0 - Critical
+# Severity 1 - Error
+# Severity 2 - Warning
+# Severity 3 - Informational
+# Severity 4 - Verbose
+
+locals {
+  database_cpu_threshold = 70
+  database_mem_threshold = 85
+}
+
+resource "azurerm_monitor_metric_alert" "db_cpu_util" {
+  count = local.alerting_enabled
+
+  name                = "${var.environment}-db-cpu"
+  description         = "${var.environment} database CPU utilization is greater than ${local.database_cpu_threshold}%"
+  resource_group_name = var.resource_group
+  scopes              = [data.azurerm_postgresql_server.postgres_server.id]
+  frequency           = "PT1M"
+  window_size         = "PT5M"
+  severity            = 1
+
+  criteria {
+    aggregation      = "Average"
+    metric_name      = "cpu_percent"
+    metric_namespace = "Microsoft.DBforPostgreSQL/servers"
+    operator         = "GreaterThanOrEqual"
+    threshold        = local.database_cpu_threshold
+  }
+
+  action {
+    action_group_id = azurerm_monitor_action_group.action_group[0].id
+  }
+}
+
+resource "azurerm_monitor_metric_alert" "db_mem_util" {
+  count = local.alerting_enabled
+
+  name                = "${var.environment}-db-mem"
+  description         = "${var.environment} database memory utilization is greater than ${local.database_mem_threshold}%"
+  resource_group_name = var.resource_group
+  scopes              = [data.azurerm_postgresql_server.postgres_server.id]
+  frequency           = "PT1M"
+  window_size         = "PT5M"
+  severity            = 1
+
+  criteria {
+    aggregation      = "Average"
+    metric_name      = "memory_percent"
+    metric_namespace = "Microsoft.DBforPostgreSQL/servers"
+    operator         = "GreaterThanOrEqual"
+    threshold        = local.database_mem_threshold
+  }
+
+  action {
+    action_group_id = azurerm_monitor_action_group.action_group[0].id
+  }
+}
+
+resource "azurerm_monitor_scheduled_query_rules_alert" "db_query_duration" {
+  count = local.alerting_enabled
+
+  name                = "${var.environment}-db-query-duration"
+  description         = "${var.environment} DB query durations >= 10s"
+  location            = var.location
+  resource_group_name = var.resource_group
+  severity            = 2
+  frequency           = 5
+  time_window         = 5
+
+  data_source_id = azurerm_application_insights.app_insights.id
+
+  query = <<-QUERY
+dependencies
+| where timestamp >= ago(5m) and name has "SQL:" and duration >= 10000
```

Comment:
Worried that this trigger in normal operation. Suggest 60 seconds.

Reply:
Yeah, 10 seems too low.   60 might be too high for Azure function queries, which seem to get killed at 30 seconds.

Reply:
My thinking was it should be under 30 seconds, due to Azure functions getting killed at the 30 second mark.

The severity of the alert is also set a warn, so PagerDuty will send an email but not page.